### PR TITLE
#57: Query Builder (closes #57)

### LINKED ISSUE #57: Query Builder
[Trello card](https://trello.com/c/586b8e774908ac4d871c281c/)

## requirements

getCustomers() already accepts a `query` param, that must conform to a complex JSON schema. For simple use cases, we should provide a helper function `ch.buildQuery` that will return the correct `query` object.

We want to cover the following two use cases:

* query all customers where `param1 = foo && param2 = bar && ...`
* query all customers where `param1 == foo || param2 == bar || ...`

## specs

write a helper function with the following signature, or a similar one:

```js
const buildQuery = (query: {
  type: 'AND' | 'OR',
  matches: Array<{[key: string]: string}>
}): Query => { ... }

```

### PR DESCRIPTION
**Pull request by ascariandrea**
Closes #57

### tests performed
happy CI

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: test/e2e/customer.e2e.js [hunk 1]
```diff
@@ -150,26 +155,59 @@ describe('ContactHub', () => {
...
+        .combinedQuery(combinedQuery)
+        .build();
+
+      const customers = await ch.getCustomers({ query });
+
+      customers.forEach(c => {
+        expect(c.base).toBeTruthy();
+        expect(c.base && c.base.firstName !== null).toBe(true);
+        expect(c.base && c.base.lastName !== null).toBe(true);
+        expect(c.base && c.base.contacts && (c.base.contacts.email || c.base.contacts.phone)).toBeTruthy(); // eslint-disable-line max-len
```
Comment (dhinus) [inline test/e2e/customer.e2e.js]: `eslint-disable-line max-len` can now be removed.

File: src/types.js [hunk 2]
```diff
@@ -6,9 +6,49 @@ export type Auth = {
...
+    condition: AtomicCondition | CompositeCondition,
+    timeframe: AtomicCondition | CompositeCondition
+  }
+};
+
+type CombinedQueryConjuction = 'INTERSECT' | 'UNION' | 'EXCEPT';
+
+export type CombinedQuery = {
+  type: string, // pattern ^combined$
+  conjuction: CombinedQueryConjuction,
```
Comment (dhinus) [inline src/types.js]: `s/conjuction/conjunction/` (here and in a few other places)

File: README.md [hunk 3]
```diff
@@ -542,6 +542,49 @@ ch.getCustomers({
...
+    new combinedQuery()
+      .conjunction('INTERSECT')
+      .addQuery(nameQuery)
+      .addQuery(emailWithExample)
+      .build()
+  )
+  .build();
+```
+
+The `createQuery` method from `ContactHub` class is for create simple query with an `AtomicCondition`.
```
Comment (dhinus) [inline README.md]: `s/for create/for creating/`

File: src/types.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ type CombinedQueryConjuction = 'INTERSECT' | 'UNION' | 'EXCEPT';
 
 export type CombinedQuery = {
   type: string, // pattern ^combined$
-  conjuction: CombinedQueryConjuction,
+  conjunction: CombinedQueryConjuction,
```
Comment (dhinus) [inline src/types.js]: ...same error on the right-hand side as well 😄 
`CombinedQueryConjuction` -> `CombinedQueryConjunction`

**Comment by ascariandrea** [comment]
Fixed!

**Pull request merged**